File: testdata/jennies/builders/anonymous_struct/TypescriptBuilder/src/anonymous_struct/somestruct_builder_gen.ts
import * as cog from '../cog';
import * as anonymous_struct from '../anonymous_struct';

export class SomeStructBuilder implements cog.Builder<anonymous_struct.SomeStruct> {
    private readonly internal: anonymous_struct.SomeStruct;

    constructor() {
        this.internal = anonymous_struct.defaultSomeStruct();
    }

    build(): anonymous_struct.SomeStruct {
        return this.internal;
    }

    time(time: {
	from: string;
	to: string;
}): this {
        this.internal.time = time;
        return this;
    }
}

File: testdata/jennies/builders/anonymous_struct/GoBuilder/anonymous_struct/somestruct_builder_gen.go
package anonymous_struct

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomeStruct] = (*SomeStructBuilder)(nil)

type SomeStructBuilder struct {
    internal *SomeStruct
    errors cog.BuildErrors
}

func NewSomeStructBuilder() *SomeStructBuilder {
	resource := NewSomeStruct()
	builder := &SomeStructBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomeStructBuilder) Build() (SomeStruct, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomeStruct{}, err
	}

	return *builder.internal, cog.MakeBuildErrors("anonymous_struct.someStruct", builder.errors)
}

func (builder *SomeStructBuilder) Time(time struct {
    From string `json:"from"`
    To string `json:"to"`
}) *SomeStructBuilder {
    builder.internal.Time = &time

    return builder
}

